demo/server.js

```diff
@@ -150,4 +150,46 @@ initSqlJs().then(SQL => {
   });
 
   app.listen(port, () => console.log(`Listening on port ${port}`));
+
+   // ACCURACY RECORD MANAGEMENT
+   function presistAccuracyRecord(accuracy) {
+    db.run('INSERT INTO accuracy VALUES (?, ?, ?, ?, CURRENT_TIMESTAMP);', [
+      accuracy.transactionHash,
+      accuracy.blockNumber,
+      accuracy.modelId,
+      accuracy.accuracy,
+    ]);
+    fs.writeFile(dbPath, Buffer.from(db.export()), () => {});
+  }
+  
+ // Add a new accuracy record for a model
+  app.post('/api/accuracy', jsonParser, (req, res) => {
+    const body = req.body
+    presistAccuracyRecord(body);
+    return res.sendStatus(200);
+  });
+
+  // Get the accuracy history 
+  app.get('/api/accuracy/model', (req, res) => {
+    const { modelId  } = req.query;
+    if (modelId != null) {
+      const str = 'SELECT * FROM accuracy where model_id == ' +modelId+' ORDER BY timestamp;';
+      const getStmt = db.prepare(str);
+      const accuracyHistory = [];
+    while (getStmt.step()) {
+      const accuracy = getStmt.get();
+      accuracyHistory.push({
+        transactionHash: accuracy[0],
+        blockNumber: accuracy[1],
+        modelId: accuracy[2],
+        accuracy: accuracy[3],
+        timestamp: accuracy[4],
+      });
+    }
+    getStmt.free();
+    res.send({ accuracyHistory});
+    } else {
+      return res.status(400).send({ message: "Not found." });
```

Comment:
I guess you copied this line from elsewhere in the file but the message is not really true. The problem was that the modelId was not given in the request.

Reply:
Well it could have been given as `null` but the modelId was not given as a valid value.

Reply:
> I guess you copied this line from elsewhere in the file but the message is not really true. The problem was that the modelId was not given in the request.

yeah this copied from your above code to find model by modelled or address , I change it to null